recipes/cyclonedds/all/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.15)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+if (${MSVC})
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
It's not reflected in the recipe. If it requires 17, the recipe should check it too.